columnflow/plotting/plot_all.py

```diff
@@ -8,51 +8,154 @@
+                    raise RuntimeError(
+                        f"shift {shift_inst} found in histogram but {shift_pairs[shift_inst]} is missing; "
+                        f"existing shifts: {','.join(map(str, list(shift_ax)))}",
+                    )
+                shift_name = shift_inst.name
+            else:
+                shift_name = nominal_shift.name
+            # store the slice
+            _h = _h[{"shift": hist.loc(shift_name)}]
+            if shift_inst not in shift_stacks:
+                shift_stacks[shift_inst] = _h
+            else:
+                shift_stacks[shift_inst] += _h
+
+    # loop over bins, subtract nominal yields from stacked yields and merge differences into
+    # a systematic error per bin using the given method (quadratic sum vs. evelope)
+    # note 1: if the up/down variations of the same shift source point in the same direction, a
+    #         statistical combination is pointless and their minimum/maximum is selected instead
+    # note 2: relative signs are consumed into the meaning of "up" and "down" here as they already
+    #         are combinations evaluated for a specific direction
+    syst_error_up = []
+    syst_error_down = []
+    for b in range(h.axes[0].size):
+        up_diffs = []
+        down_diffs = []
+        for source, (up_shift, down_shift) in shift_groups.items():
+            up_diff = shift_stacks[up_shift].values()[b] - h.values()[b]
+            down_diff = shift_stacks[down_shift].values()[b] - h.values()[b]
+            up_diffs.append(max(up_diff, down_diff, 0))
+            down_diffs.append(min(up_diff, down_diff, 0))
+        # combination based on the method
+        if method == "quadratic_sum":
+            up_diff = sum(d**2 for d in up_diffs)**0.5
+            down_diff = sum(d**2 for d in down_diffs)**0.5
+        else:  # envelope
+            up_diff = max(up_diffs)
+            down_diff = min(down_diffs)
+        # save values
+        syst_error_up.append(up_diff)
+        syst_error_down.append(down_diff)
+
+    # compute relative systematic errors
+    rel_syst_error_up = np.array(syst_error_up) / h.values()
+    rel_syst_error_up[np.isnan(rel_syst_error_up)] = 0.0
+    rel_syst_error_down = np.array(syst_error_down) / h.values()
+    rel_syst_error_down[np.isnan(rel_syst_error_down)] = 0.0
+
+    # compute the baseline
+    # fill 1 in places where both numerator and denominator are 0, and 0 for remaining nan's
+    baseline = h.values() / norm
+    baseline[(h.values() == 0) & (norm == 0)] = 1.0
+    baseline[np.isnan(baseline)] = 0.0
+
+    bar_kwargs = {
+        "x": h.axes[0].centers,
+        "bottom": baseline * (1 - rel_syst_error_down),
+        "height": baseline * (rel_syst_error_up + rel_syst_error_down),
+        **kwargs,
+        "width": h.axes[0].edges[1:] - h.axes[0].edges[:-1],
+        "hatch": "\\\\\\",
+        "linewidth": 0,
+        "color": "none",
+        "edgecolor": "#30c300",
+        "alpha": 1.0,
+    }
```

Comment:
To give priority to values defined in `**kwargs`
```suggestion
    bar_kwargs = {
        "x": h.axes[0].centers,
        "bottom": baseline * (1 - rel_syst_error_down),
        "height": baseline * (rel_syst_error_up + rel_syst_error_down),
        "width": h.axes[0].edges[1:] - h.axes[0].edges[:-1],
        "hatch": "\\\\\\",
        "linewidth": 0,
        "color": "none",
        "edgecolor": "#30c300",
        "alpha": 1.0,
        **kwargs,
    }
```